### DISCUSSION TITLE
IP地址已经变动但是返回addresses unchanged

### DISCUSSION THREAD

**Discussion by SnowRain1326** [jeessy2/ddns-go#1163] [Q&A] [answered]
明确IP地址已经变动了但是返回提示没变动

**Comment by jeessy2** [comment]
配置/日志截图

**Reply by SnowRain1326** [reply]
![Uploading 屏幕截图 2024-02-28 201834.png…]()

**Reply by jeessy2** [reply]
没传起

**Reply by SnowRain1326** [reply]
![屏幕截图_18-6-2024_144154_127 0 0 1](https://github.com/jeessy2/ddns-go/assets/26730782/ea6bc547-723b-4637-a751-d71c161d9b3f)
![屏幕截图 2024-02-28 201834](https://github.com/jeessy2/ddns-go/assets/26730782/e322fadd-f18b-46e7-b4fd-11cb82f0c050)

**Reply by jeessy2** [reply]
这个应该是dynv6返回给你的，地址未改变。直接去dynv6看地址是不是日志中的v4/v6地址

**Reply by SnowRain1326** [reply]
第二张截图里面最上面那个白色的部分就是dynv6的截图，v4没变但是v6已经跟本地的不一样了，但是还是返回的address unchanged

**Reply by jeessy2** [reply]
你设置的为`@2`，用第二个v6

**Reply by SnowRain1326** [reply]
第二个v6的结尾是a676，也是和dynv6上的不一样的

**Comment by jeessy2** [comment]
14:29:30不是写了调用成功了？如果还是不成功，因该是配置没弄好

**Reply by SnowRain1326** [reply]
调用成功了，但是即使地址发生变化也不更新，只返回address unchanged，就有点不知道错在哪里
url用的http://dynv6.com/api/update?zone=你的域名&token=对应密钥&ipv4&ipv6prefix
如果在v4/v6后面加上#={ip}的话就会报错invalid address

**Reply by WaterLemons2k** [reply]
> 调用成功了，但是即使地址发生变化也不更新，只返回address unchanged，就有点不知道错在哪里 url用的[http://dynv6.com/api/update?zone=你的域名&token=对应密钥&ipv4&ipv6prefix](http://dynv6.com/api/update?zone=%E4%BD%A0%E7%9A%84%E5%9F%9F%E5%90%8D&token=%E5%AF%B9%E5%BA%94%E5%AF%86%E9%92%A5&ipv4&ipv6prefix) 如果在v4/v6后面加上#={ip}的话就会报错invalid address

因为每次调用时只会传进去**一个** IPv4/IPv6 地址。需要创建两个配置：一个更新 IPv4，另一个更新 IPv6。

**Reply by jeessy2** [reply]
对。要建二个，dynv6的ipv4/ipv6参数不同

**Reply by jeessy2** [reply, marked as answer]
参考刚更新的文档：https://github.com/jeessy2/ddns-go/wiki/Callback%E9%85%8D%E7%BD%AE%E5%8F%82%E8%80%83

**Reply by SnowRain1326** [reply]
先生真乃神医！
分别建了两个配置来更新之后就正常了，完全没有想到是这个原因
http://dynv6.com/api/update?zone=你的域名&token=对应密钥&ipv4=#{ip}
http://dynv6.com/api/update?zone=你的域名&token=对应密钥&ipv6prefix=#{ip}

**Reply by mmyddd** [reply]
建了两个配置之后怎么一起填到url里面呢
